framework/include/interfacekernels/InterfaceKernel.h

```diff
@@ -53,6 +53,46 @@ class InterfaceKernel : public DGKernel
 
   const VariableValue & _neighbor_value;
   const VariableGradient & _grad_neighbor_value;
+
+  /// Vector of MooseEnums specifying whether residual save-in aux variables correspond to master or slave side
+  std::vector<MooseEnum> _save_in_var_side;
+
+  /** The names of the aux variables that will be used to save-in residuals
+   * (includes both master and slave variable names)
+   */
+  std::vector<AuxVariableName> _save_in_strings;
+
+  /// Whether there are master residual aux variables
+  bool _has_master_residuals_saved_in;
+
+  /// The aux variables to save the master residual contributions to
+  std::vector<MooseVariable *> _master_save_in_residual_variables;
+
+  /// Whether there are slave residual aux variables
+  bool _has_slave_residuals_saved_in;
+
+  /// The aux variables to save the slave contributions to
+  std::vector<MooseVariable *> _slave_save_in_residual_variables;
+
+  /// Vector of MooseEnums specifying whether jacobian save-in aux variables correspond to master or slave side
+  std::vector<MooseEnum> _diag_save_in_var_side;
```

Comment:
So I recommend that you don't create a vector of MooseEnums. The reason is that MooseEnums don't behave well if you try to push_back or extend the vector. We created `MultiMooseEnum` for this purpose instead.